src/gtk/textctrl.cpp

```diff
@@ -1017,6 +1023,52 @@ void wxTextCtrl::GTKSetJustification()
     }
 }
 
+#if wxUSE_SPELLCHECK
+
+bool wxTextCtrl::EnableSpellCheck(const bool enable)
+{
+    wxCHECK_MSG( IsMultiLine(), false, 
+                "Unable to enable spell check on control "
+                "which does not have wxTE_MULTILINE style" );
+
+    GtkTextView *textview = GTK_TEXT_VIEW(m_text);
+    wxCHECK_MSG( textview, false, wxS("wxTextCtrl is not a GtkTextView" ));
+
+    wxString lang = wxLocale::GetLanguageCanonicalName(wxLANGUAGE_DEFAULT);
+
+    GtkSpellChecker *spell = gtk_spell_checker_get_from_text_view(textview);
+
+    if ( enable )
+    {
+        if ( !spell )
+        {
+            spell = gtk_spell_checker_new();
+            gtk_spell_checker_attach(spell, textview);
+        }
+
+        if ( !gtk_spell_checker_set_language(spell, NULL, NULL) )
+            return false;
+    }
+    else
+    {
+        if ( spell )
+            gtk_spell_checker_detach(spell);
+    }
+
+   return true;
+}
+
+bool wxTextCtrl::IsSpellCheckEnabled() const
+{
```

Comment:
It looks like this ought to check for `IsMultiLine()` too -- and maybe just return `false`? As it is, we're probably going to crash using null `textview` if this is called for a single line control.

Reply:
I have added a guard for both IsMultiLine() and textview == NULL.